benches/benches/bevy_ecs/iteration/heavy_compute.rs

```diff
@@ -20,7 +20,7 @@ pub fn heavy_compute(c: &mut Criterion) {
     group.warm_up_time(std::time::Duration::from_millis(500));
     group.measurement_time(std::time::Duration::from_secs(4));
     group.bench_function("base", |b| {
-        ComputeTaskPool::init(TaskPool::default);
+        ComputeTaskPool::get_or_init(TaskPool::default);
```

Comment:
Half of the calls are with the default value, maybe add a `.get_or_init_default` method?

Reply:
My intuition is that a lot of these calls could just be replaced with calls to the panicking getter, but you would have to look at the context of each call to be sure, which is probably out of the scope of this PR.

Regardless, I don't think that a default initializer should be added, as [TaskPoolOptions](https://github.com/bevyengine/bevy/blob/f9e50e767b76382527bff93385247829461758a0/crates/bevy_core/src/task_pool_options.rs) from `bevy_core` is the preferred way of initializing the global taskpools. We should be encouraging using that, and discouraging using `TaskPool::default`.

`TaskPoolOptions` could potentially be moved from `bevy_core` to `bevy_tasks` to facilitate using that instead of `TaskPool::default`, but that is definitely outside of the scope of this PR.